### ISSUE TITLE
Error 4294967295 after restarting the first time I open a WSL terminal 

### ISSUE THREAD

**Issue by ioweb-gr** [microsoft/WSL#5592]
# Environment

```
Windows build number: Microsoft Windows [Version 10.0.19042.388]
Your Distribution version: Release:        18.04
Whether the issue is on WSL 2 and/or WSL 1: Both
```

# Steps to reproduce

It's been a couple builds now that after a restart, the first time I open a WSL terminal I get the following error. 
```
The I/O operation has been aborted because of either a thread exit or an application request.

[process exited with code 4294967295]

```

#  Expected behavior

The terminal starts correctly

# Actual behavior

The terminal does not start and is throwing an error 

The second terminal works. The first one takes a long time to start and fails

**Comment by gregdegruy** [comment]
I am also facing this issue. Ubunut 18.04 works fine, but the Ubuntu app from the Microsoft Store gives me this error.

```
OS Name:                   Microsoft Windows 10 Pro Insider Preview
OS Version:                10.0.20180 N/A Build 20180
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
System Manufacturer:       MSI
System Model:              MS-7A11
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 94 Stepping 3 GenuineIntel ~4008 Mhz
BIOS Version:              American Megatrends Inc. 2.50, 7/24/2016
Windows Directory:         C:\WINDOWS
System Directory:          C:\WINDOWS\system32
Boot Device:               \Device\HarddiskVolume2
System Locale:             en-us;English (United States)
Input Locale:              ja;Japanese
Time Zone:                 (UTC-08:00) Pacific Time (US & Canada)
```

**Comment by simis2626** [comment]
have a look at this thread, solved my issue with the workaround there.

TL;DR; looks like often a too long PATH env var is causing the issue (there's a fix inbound)
https://github.com/microsoft/WSL/issues/5633

**Comment by lingr7** [comment]
WSL2 出现“参考的对象类型不支持尝试的操作”https://zhuanlan.zhihu.com/p/151392411
我遇到的问题，搜索到了这里，实际解决方案是
powershell Administrator.
```
netsh winsock reset
```
restart system

**Comment by CacaoRick** [comment]
This work for me, no need restart system (Run as Administrator):
```
netsh winsock reset
```
But don't know why every time I start my computer, I need run this command again.

**Comment by lingr7** [comment]
> This work for me, no need restart system (Run as Administrator):
> 
> ```
> netsh winsock reset
> ```
> 
> But don't know why every time I start my computer, I need run this command again.

for me, it is about the program **docker**. docker change or use some wsl api and modify enviroment.
when i update docker and windows os version. the issue do not exist again. 
maybe you need remove docker,  update docker  or use docker often to make its changes be checked or kept.
